numpy/_core/tests/test_regression.py

```diff
@@ -2622,3 +2622,16 @@ def test_vectorize_fixed_width_string(self):
         f = str.casefold
         res = np.vectorize(f, otypes=[arr.dtype])(arr)
         assert res.dtype == "U30"
+
+    def test_repeated_square_consistency(self):
+        # gh-26940
+        buf = np.array([-5.171866611150749e-07 + 2.5618634555957426e-07j,
+                        0, 0, 0, 0, 0])
+        in_vec = buf[:3]
+        # Output vector immediately follows input vector
+        # to reproduce off-by-one in nomemoverlap check.
+        res0 = buf[3:]
```

Comment:
Part of the problem here is reversed arrays.  Could you add parametrization to reverse the first, second, neither and both (both may end up reversed in the inner loop, but maybe still good).

Reply:
Done.

Btw the check with non-reversed input and reversed output after it also failed (with `-ffp-contract=off`), and the ones with the reversed input being first didn't (they would had if the output was adjacent before the input).